### ISSUE TITLE
public CI with github status

### ISSUE THREAD

**Issue by jldec** [projectriff/riff#258]
### workflow
- push a commit
- see "in progress" github status 
- wait reasonably short time for status ok (green) or fail (red) 
- click through to CI log

### questions / caveats
- work with PRs on forks (can't expose credential to code in fork)?
- cross-repo dependencies
- separate out image build (and push to dockerhub)

**Comment by jchesterpivotal** [comment]
As I see it:

* [ ] Move repo to projectriff org (cred-alert comes back clean)
    - [ ] Change git resource definitions in pipelines
    - [ ] Change authorised team(s) to `projectriff/riff-dev` and `projectriff/riff-watchers`
* [ ] Audit tasks for logging of credentials
* [ ] Backup database
* [ ] Make fresh deployment without history
* [ ] Open it up
* [ ] Get the domain/URI and certificate sorted out

**Comment by jchesterpivotal** [comment]
Added github status resource to pipeline, which tells Github about pending, failed and succeeded statuses per-commit.

**Comment by jchesterpivotal** [comment]
@jldec I think I've got the "workflow" bit done. I might suggest we split out PRs as another issue. Anything else you'd need before we close this one?

**Comment by jchesterpivotal** [comment]
Actually, scratch that, I have a bit more to do. It currently only applies status on the build-container jobs at the start of the pipeline. It'd be nice if it also updated it for other jobs, especially system tests.

**Comment by jchesterpivotal** [comment]
Last todos:

- [ ] Add status to the acceptance test build
- [ ] Add statuses to riff-pr
- [ ] Add statuses to dockerimages

**Comment by trisberg** [comment]
should we move the `pivotal-cf/pfs-system-test` to `projectriff\acceptance-test`?

change the `sk8s` docker hub user to `riffci` for images we use

**Comment by jchesterpivotal** [comment]
No objection, but this issue has been fairly busy and I'm anxious to close it in its own right.

**Comment by jchesterpivotal** [comment]
Except, actually, you're right about both of those being private bits.

Ugh.

**Comment by jchesterpivotal** [comment]
Infinite todos

- [x] ~~Add status to the acceptance test build~~ can't be done with our current setup
- [ ] Status for helm charts
- [ ] Add statuses to riff-pr
- [ ] Add statuses to dockerimages

**Comment by jchesterpivotal** [comment]
Actually, I have a hunch that I am wrong about not being able to set nice commit statuses for every job / commit permutation. The theory would be:

1. Set the status for _every_ commit in _every_ job.
2. Rely on the `context` key to distinguish what is going on (so a commit might be passed for `ci/riff/build-http-gateway-container` but pending on `run-riff-system-tests-gke`).
3. Use `version: every` to ensure we are testing all the permutations. Right now we don't.

**Comment by jchesterpivotal** [comment]
blocking on projectriff/ci#50

**Comment by jchesterpivotal** [comment]
I'm going to call this done. We have a public CI, images are pushed to public repos and so on. I'll break out missing FATS tests into another issue.
